extra/mariabackup/xtrabackup.cc

```diff
@@ -4661,7 +4669,10 @@ static bool xtrabackup_backup_func()
 		checkpoint_lsn_start = log_sys.log.get_lsn();
 		checkpoint_no_start = log_sys.next_checkpoint_no;
 
-		log_sys.log.read(max_cp_field, {buf, OS_FILE_LOG_BLOCK_SIZE});
+		if (log_sys.log.read(max_cp_field, {buf, OS_FILE_LOG_BLOCK_SIZE})) {
+			mysql_mutex_unlock(&log_sys.mutex);
+			goto fail;
+		}
 
 		if (checkpoint_no_start
```

Comment:
We could simplify the above by:
```c++
err = log_sys.log.read(…);
if (err != DB_SUCCESS);
else if (checkpoint_no_start …
```
In this way, we would also avoid duplicating the call to `mysql_mutex_unlock(&log_sys.mutex)`.